### ISSUE TITLE
Building an imgui branch

### ISSUE THREAD

**Issue by pixtur** [mellinoe/ImGui.NET#95]
You mentioned [here](https://mellinoe.github.io/graphics/2018/10/12/imgui-net-overhaul.html) that it would be possible to build imgui.net for branches of dear imgui. Would it be possible to give some pointers on where to start if I wanted to build the docking branch?

**Comment by mellinoe** [comment]
Here's the steps I've followed to use the docking branch (as of a month or so ago):

* Clone [cimgui](https://github.com/cimgui/cimgui), and update its `imgui` submodule to the docking branch.
* Build cimgui (using cmake / Visual Studio -- follow the instructions on their repo).
* Copy the produced cimgui.dll into your ImGui.NET repo (`<root>/deps/cimgui/win-x64/`, as appropriate)
* Run `cimgui\generator\generator.bat` (follow the instructions there if it doesn't work)
* Copy `cimgui\generator\output\structs_and_enums.json` and `cimgui\generator\output\definitions.json` into `ImGui.NET\src\CodeGenerator`
* Re-run the CodeGenerator program, passing the "Generated" folder as the first argument:
`dotnet run -p src\CodeGenerator\CodeGenerator.csproj src\ImGui.NET\Generated`
* Re-build ImGui.NET and use it with the new cimgui.dll.

**Comment by wihu** [comment]
I tried building docking branch but I get Entry not found exception when calling either IsWindowDocked or SetNextWindowDockID.
I also tried to dump the cimgui.dylib entry points:
[cimgui_entries.txt](https://github.com/mellinoe/ImGui.NET/files/3038136/cimgui_entries.txt)
and seems those functions have _ZXX prefix, any idea why these functions are exported differently?

**Comment by sonoro1234** [comment]
generator2.bat should be used before cimgui compilation (not afterwards)

**Comment by wihu** [comment]
yes, those instructions should be swapped i.e. run generator2 and then build cimgui.

**Comment by pixtur** [comment]
I just noticed that I never thanked you for this quick answer.
